src/Uno.UI/UI/Xaml/Controls/ItemsControl/ItemsControl.cs

```diff
@@ -1115,13 +1101,53 @@ internal DependencyObject GetContainerForIndex(int index)
 		{
 			var item = ItemFromIndex(index);
 
-			var container = IsItemItsOwnContainerOverride(item)
-				? item as DependencyObject
-				: GetContainerForItemOverride();
+			DependencyObject elementTemplateRoot = DataTemplateHelper.ResolveTemplate(ItemTemplate, ItemTemplateSelector, item, null)?.LoadContentCached() as DependencyObject;
+
+			var isItemItsOwnContainer = IsItemItsOwnContainerOverride(item);
+			var isContainerFromTemplateRoot = !isItemItsOwnContainer
+				&& this is Primitives.Selector
```

Comment:
Codacy found an issue: [Offload the code that's conditional on this 'is' test to the appropriate subclass and remove the test.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5961910)